### ISSUE TITLE
JanusGraph Server 0.6.3 only logging warnings

### ISSUE THREAD

**Issue by FlorianHockmann** [JanusGraph/janusgraph#3681]
It looks like the default logging config has changed in JanusGraph 0.6.3. This is how the logs look like for that version:

```bash
> docker run --rm -it janusgraph/janusgraph:0.6.3
/etc/opt/janusgraph/janusgraph-server.yaml will be used to start JanusGraph Server in foreground
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/opt/janusgraph/lib/logback-classic-1.2.11.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/janusgraph/lib/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
07:30:12 WARN  org.apache.tinkerpop.gremlin.server.AbstractChannelizer - The org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0 serialization class is deprecated.
07:30:12 WARN  org.apache.tinkerpop.gremlin.server.AbstractChannelizer - The org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0 serialization class is deprecated.
```

So, the usual logs, including the JanusGraph ASCII-art, information about the used config, and the fact that it started successfully and is now listening on port 8182 are completely missing here. We only see the SLF4J logs and two warnings about deprecated Gryo message serializers being configured.

Compare that with logs from JanusGraph 0.6.2:

```bash
> docker run --rm -it janusgraph/janusgraph:0.6.2
/etc/opt/janusgraph/janusgraph-server.yaml will be used to start JanusGraph Server in foreground
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/opt/janusgraph/lib/slf4j-log4j12-1.7.30.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/janusgraph/lib/logback-classic-1.1.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
0    [main] INFO  org.janusgraph.graphdb.server.JanusGraphServer  -
   mmm                                mmm                       #
     #   mmm   m mm   m   m   mmm   m"   "  m mm   mmm   mmmm   # mm
     #  "   #  #"  #  #   #  #   "  #   mm  #"  " "   #  #" "#  #"  #
     #  m"""#  #   #  #   #   """m  #    #  #     m"""#  #   #  #   #
 "mmm"  "mm"#  #   #  "mm"#  "mmm"   "mmm"  #     "mm"#  ##m#"  #   #
                                                         #
                                                         "

78   [main] INFO  com.jcabi.manifests.Manifests  - 115 attributes loaded from 330 stream(s) in 61ms, 115 saved, 4876 ignored: ["Agent-Class", "Ant-Version", "Archiver-Version", "Automatic-Module-Name", "Bnd-LastModified", "Build-Date", "Build-Host", "Build-Id", "Build-Java-Version", "Build-Jdk", "Build-Jdk-Spec", "Build-Number", "Build-Tag", "Build-Version", "Built-By", "Built-JDK", "Built-OS", "Built-Status", "Bundle-ActivationPolicy", "Bundle-Activator", "Bundle-Category", "Bundle-ClassPath", "Bundle-Classpath", "Bundle-ContactAddress", "Bundle-Copyright", "Bundle-Description", "Bundle-DocURL", "Bundle-License", "Bundle-ManifestVersion", "Bundle-Name", "Bundle-NativeCode", "Bundle-RequiredExecutionEnvironment", "Bundle-SymbolicName", "Bundle-Vendor", "Bundle-Version", "Change", "Class-Path", "Copyright", "Created-By", "DSTAMP", "Dependencies", "DynamicImport-Package", "Eclipse-BuddyPolicy", "Eclipse-ExtensibleAPI", "Embed-Dependency", "Embed-Transitive", "Export-Package", "Extension-Name", "Extension-name", "Fragment-Host", "Gradle-Version", "Gremlin-Lib-Paths", "Gremlin-Plugin-Dependencies", "Gremlin-Plugin-Paths", "Ignore-Package", "Implementation-Build", "Implementation-Build-Date", "Implementation-Title", "Implementation-URL", "Implementation-Vendor", "Implementation-Vendor-Id", "Implementation-Version", "Import-Package", "Include-Resource", "JCabi-Build", "JCabi-Date", "JCabi-Version", "Java-Vendor", "Java-Version", "Main-Class", "Manifest-Version", "Module-Origin", "Module-Requires", "Multi-Release", "Originally-Created-By", "Os-Arch", "Os-Name", "Os-Version", "Package", "Premain-Class", "Private-Package", "Provide-Capability", "Require-Bundle", "Require-Capability", "SCM-Revision", "Scm-Connection", "Scm-Revision", "Scm-Url", "Source-Compatibility", "Specification-Title", "Specification-Vendor", "Specification-Version", "TODAY", "TSTAMP", "Target-Compatibility", "Timestamp", "Tool", "Vcs-Release-Hash", "Vcs-Release-Tag", "X-Compile-Elasticsearch-Snapshot", "X-Compile-Elasticsearch-Version", "X-Compile-Lucene-Version", "X-Compile-Source-JDK", "X-Compile-Target-JDK", "artifactId", "groupId", "hash", "implementation-version", "janusgraphVersion", "mode", "package", "service", "tinkerpopVersion", "url", "version"]
79   [main] INFO  org.janusgraph.graphdb.server.JanusGraphServer  - JanusGraph Version: 0.6.2
79   [main] INFO  org.janusgraph.graphdb.server.JanusGraphServer  - TinkerPop Version: 3.5.3
81   [main] INFO  org.janusgraph.graphdb.server.JanusGraphServer  - Configuring JanusGraph Server from /etc/opt/janusgraph/janusgraph-server.yaml
247  [main] INFO  org.apache.tinkerpop.gremlin.server.util.MetricManager  - Configured Metrics CsvReporter configured with report interval=180000ms to fileName=/tmp/gremlin-server-metrics.csv
300  [main] INFO  org.apache.tinkerpop.gremlin.server.util.MetricManager  - Configured Metrics JmxReporter configured with domain= and agentId=
305  [main] INFO  org.apache.tinkerpop.gremlin.server.util.MetricManager  - Configured Metrics Slf4jReporter configured with interval=180000ms and loggerName=org.apache.tinkerpop.gremlin.server.Settings$Slf4jReporterMetrics
949  [main] INFO  org.janusgraph.diskstorage.configuration.builder.ReadConfigurationBuilder  - Set default timestamp provider MICRO
1059 [main] INFO  org.janusgraph.graphdb.idmanagement.UniqueInstanceIdRetriever  - Generated unique-instance-id=c0a801011-66fc746a7d1e1
1091 [main] INFO  org.janusgraph.diskstorage.Backend  - Configuring index [search]
1164 [main] INFO  org.janusgraph.diskstorage.configuration.ExecutorServiceBuilder  - Initiated fixed thread pool of size 24
1217 [main] INFO  org.janusgraph.diskstorage.log.kcvs.KCVSLog  - Loaded unidentified ReadMarker start time 2023-03-24T07:32:52.993Z into org.janusgraph.diskstorage.log.kcvs.KCVSLog$MessagePuller@412c995d
1218 [main] INFO  org.apache.tinkerpop.gremlin.server.util.DefaultGraphManager  - Graph [graph] was successfully configured via [/etc/opt/janusgraph/janusgraph.properties].
1218 [main] INFO  org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor  - Initialized Gremlin thread pool.  Threads in pool named with pattern gremlin-*
1256 [main] INFO  org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor  - Initialized GremlinExecutor and preparing GremlinScriptEngines instances.
3614 [main] INFO  org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor  - Initialized gremlin-groovy GremlinScriptEngine and registered metrics
3617 [main] INFO  org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor  - A GraphTraversalSource is now bound to [g] with graphtraversalsource[standardjanusgraph[berkeleyje:/var/lib/janusgraph/data], standard]
3629 [main] INFO  org.apache.tinkerpop.gremlin.server.op.OpLoader  - Adding the standard OpProcessor.
3679 [main] INFO  org.apache.tinkerpop.gremlin.server.op.OpLoader  - Adding the session OpProcessor.
3682 [main] INFO  org.apache.tinkerpop.gremlin.server.op.OpLoader  - Adding the traversal OpProcessor.
3711 [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Executing start up LifeCycleHook
3721 [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Executed once at startup of Gremlin Server.
3725 [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - idleConnectionTimeout was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
3725 [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - keepAliveInterval was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
3757 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.graphbinary-v1.0 with org.apache.tinkerpop.gremlin.driver.ser.GraphBinaryMessageSerializerV1
3757 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.graphbinary-v1.0-stringd with org.apache.tinkerpop.gremlin.driver.ser.GraphBinaryMessageSerializerV1
3758 [main] WARN  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - The org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0 serialization class is deprecated.
3802 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.gremlin-v3.0+gryo with org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0
3802 [main] WARN  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - The org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0 serialization class is deprecated.
3802 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.gremlin-v3.0+gryo-stringd with org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0
3817 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.gremlin-v3.0+json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0
3818 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0
3873 [gremlin-server-boss-1] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Gremlin Server configured with worker thread pool of 1, gremlin pool of 8 and boss thread pool of 1.
3874 [gremlin-server-boss-1] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Channel started at port 8182.
```

I would consider this a bug as it even looks like JanusGraph doesn't start successfully and instead hangs during start-up.
I already confirmed that this is not specific to the Docker images. The same behavior can be seen when the server is started locally without Docker.

For confirmed bugs, please report:

- Version: 0.6.3
- Storage Backend: inmemory
- Mixed Index Backend: none
- Expected Behavior: See all logs, including `Channel started at port 8182`
- Current Behavior: See only 2 warning logs
- Steps to Reproduce: `docker run janusgraph/janusgraph:0.6.3`

**Comment by mmarovic** [comment]
This might be related to something I encountered a while ago and reported in the mailing list:

https://lists.lfaidata.foundation/g/janusgraph-users/topic/97337825#6686

Basically, prior to 0.6.3 slf4j resolved to log4j by default, but in 0.6.3 the log4j dependency was changed to reload4j, which changed the order of resolution so now slf4j resolves to logback by default.

**Comment by FlorianHockmann** [comment]
Thanks for your feedback on this and your suggestions to fix this by using Reload4j. I just created a PR for this which does exactly that: #3745.

It looks to me like the issue was introduced by an update of Slf4j in #2967 which included a commit with the message: _relocate slf4j-log4j12 as slf4j-reload4j_. That's probably why Log4j isn't present anymore in the `lib` directory which resulted in Slf4j using Logback by default. This switch wasn't intended as it was basically a breaking change since existing Log4j config files weren't used any more. Switching to Reload4j seems like the best option to me here as it reverts this breaking change and still avoids the security issues of Log4j.

**Comment by FlorianHockmann** [comment]
Fixed in #3745.
